cerberus/tests/test_validation.py

```diff
@@ -1276,6 +1276,39 @@ def test_nested_oneof_type():
     assert_success({'nested_oneof_type': {'bar': 3}}, schema)
 
 
+def test_nested_oneofs(validator):
+    validator.schema = {'abc': {
+        'type': 'dict',
+        'oneof_schema': [
+            {'foo': {
+                'type': 'dict',
+                'schema': {'bar': {'oneof_type': ['integer', 'float']}}
+            }},
+            {'baz': {'type': 'string'}}
+        ]}}
+
+    document = {'abc': {'foo': {'bar': 'bad'}}}
+
+    expected_errors = {
+        'abc': [
+            {'oneof': [
```

Comment:
but then different of-rule-related errors (eg `oneof` and `anyof`) couldn't be discriminated.